src/System.Management.Automation/engine/runtime/Operations/NumericOps.cs

```diff
@@ -863,5 +863,34 @@ internal static object CompareIne(char lhs, char rhs)
             char secondAsUpper = char.ToUpperInvariant(rhs);
             return firstAsUpper != secondAsUpper ? Boxed.True : Boxed.False;
         }
+
+        internal static object[] Range(char start, char end)
```

Comment:
What do you think about returning `char[]` instead?

Reply:
@IISResetMe Could you please address the comment?

Reply:
I like the idea, but for now I think we should keep operator output consistent (ie. return `object[]` with individual items of the expected type).

There are are number of builtin operators that return `object[]` in this fashion (`@()`, `-split`, any comparison operator in array/filter mode etc.), changing the behavior should be a concerted effort across all of them IMO